packages/plugins/build-report/src/esbuild.ts

```diff
@@ -68,8 +67,8 @@ export const getEsbuildPlugin = (context: GlobalContext, log: Logger): PluginOpt
             const entryNames = getEntryNames(entrypoints, context);
 
             build.onEnd((result) => {
-                context.build.errors = result.errors.map((err) => err.text);
-                context.build.warnings = result.warnings.map((err) => err.text);
+                context.build.errors.push(...result.errors.map((err) => err.text));
+                context.build.warnings.push(...result.warnings.map((err) => err.text));
```

Comment:
Note: you're generating new arrays to immediately explode them
Idk if this can't be an issue for later arrays. If it is, it may be better to use forEach + push (but if it's not lgtm)